deepchecks/vision/checks/model_evaluation/image_segment_performance.py

```diff
@@ -86,18 +83,14 @@ def initialize_run(self, context: Context, dataset_kind: DatasetKind):
 
     def update(self, context: Context, batch: Batch, dataset_kind: DatasetKind):
         """Update the bins by the image properties."""
-        images = batch.images
         predictions = [tens.detach() for tens in batch.predictions]
         labels = [tens.detach() for tens in batch.labels]
+        properties_results = batch.vision_properties(batch.images, self.image_properties, PropertiesInputType.IMAGES)
```

Comment:
So, now your validation of properties happens in the initialize_run instead of the __init__. I think that this is not the right behavior - the validation should happen when its defined, if possible.
Why not validate then?

Reply:
As I see it, the general validations are the job of the batch/context and not the check

Reply:
I tend to agree with @nirhutnik. It is more elegant the way you did it (no need to call validation per check init), but the fact is that properties **can** be validated in the init, and from a user experience perspective it's better - If my properties are broken, I'll know that when creating the check rather than when running it.